test/python/transpiler/test_optimize_phase_shift_gates.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for pass marging/canceling phase-shift gates and cnot cancellation."""
+
+import numpy as np
+
+from qiskit import BasicAer, execute
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.converters import circuit_to_dag, dag_to_circuit
+from qiskit.test import QiskitTestCase
+from qiskit.transpiler.passes import OptimizePhaseShiftGates
+
+
+class TestOptimizePhaseShiftGates(QiskitTestCase):
+    """Test the TestOptimizePhaseShiftGates pass."""
+
+    def test_pass_optimize_phase_shift_gates(self):
+        """Test the cx TestOptimizePhaseShiftGates pass.
+
+        It should merge/cancel phase-shift gates and perform cnot cancellation.
+        """
+        n = 6
+        q = QuantumRegister(n, 'q')
+        circ = QuantumCircuit(q)
+        i = 0
+        while i < 150:
+            i += 1
+            x = np.random.randint(9)
```

Comment:
In general, we prefer deterministic tests.

Reply:
With "prefer" you mean I should change it now, or is that just a hint for future contributions? :)

Reply:
At some point we should move the random tests to `test/randomized` which is better suited to handle them. But until then we need them in a lot of places to increase our confidence in the code (See tests under `quantum_info` specifically). So I wouldn't remove them until they go and exist somewhere else.